datadog_checks_base/datadog_checks/base/utils/containers.py

```diff
@@ -4,19 +4,59 @@
 from six import iteritems
 
 
+class _FreezeKey(object):
+    # "Why does this class exist?"
+    # To freeze and compute hashes or mutable structures, we use sorting as an intermediary step so that the
+    # order of items in the container doesn't change the final hash (a property we could call "commutativity").
+    # When items in the container are all the same type (eg a list of strings), there is no problem to this.
+    # But when the container has items of mixed types, we'd get a `TypeError` on Python 3. (Not on Python 2 though,
+    # since comparisons there default to returning `False`.)
+    # So this class was introduced to support specific comparison cases that we need to support to fulfill the needs
+    # of integrations that use these freeze/hash helpers.
+
+    def __init__(self, value):
+        self.value = value
+
+    def __lt__(self, other):
+        # type: (_FreezeKey) -> bool
+        try:
+            lt = self.value < other.value
+        except TypeError:
+            # We're on Python 3, and values are of differing types.
+            # Some integrations may using freezing on structures that contain `None`, so we want to support it and
+            # use the same behavior than on Python 2, i.e. `None < <anything>` must return `False`...
+            if self.value is None:
+                # `None < x` -> `False`
+                return False
+            if other.value is None:
+                # `x < None` -> `True`
+                return True
+            # ...But we let other cases bubble through.
+            raise
+        else:
+            # We're on Python 2, where `a < b` never fails (returns `False` by default).
```

Comment:
We're either on py2 or on py3 wih values of the same type.

Reply:
```suggestion
            # We're on Python 2, where `a < b` never fails (returns `False` by default), or
            # we're on Python 3 and values have the same type.
```